demos/storybook/stories/drawer/with-full-config.tsx

```diff
@@ -34,69 +49,157 @@ const devices = 'Devices';
 const photos = 'Photos';
 const schedule = 'Schedule';
 const agreement = 'License Agreement';
+const monthlyReport = 'Monthly Report';
+const annualReport = 'Annual Report';
+const colorContrastGuide = 'Color Contrast Guide';
+const screenReader = 'Screen Reader';
+
+export const getIcon = (icon: string): JSX.Element | undefined => {
+    switch (icon) {
+        case '<Add />':
+            return <Add />;
+        case '<PinDrop />':
+            return <PinDrop />;
+        case '<Remove />':
+            return <Remove />;
+        case '<AddAPhoto />':
+            return <AddAPhoto />;
+        case '<Menu />':
+            return <Menu />;
+        case '<FitnessCenter />':
+            return <FitnessCenter />;
+        case '<List />':
+            return <List />;
+        case 'undefined':
+        default:
+            return undefined;
+    }
+};
+
+const headerBackgroundImageOptions = {
+    Pattern: topologyBgImage,
+    Farm: farmBgImage,
+    undefined: undefined,
+};
 
 export const withFullConfig = (context: DrawerStoryContext): StoryFnReactReturnType => {
     const drawerGroupId = 'Drawer';
     const headerGroupId = 'Header';
     const bodyGroupId = 'Body';
+    const navGroupId = 'NavGroup 1';
+    const navItemId = 'NavItem';
     const footerGroupId = 'Footer';
 
-    const open = boolean('open', true, drawerGroupId);
-    const width = number(
-        'width',
-        350,
-        {
-            range: true,
-            min: 200,
-            max: 700,
-            step: 50,
-        },
-        drawerGroupId
-    );
+    const drawerKnobs: DrawerComponentProps = {
```

Comment:
We need to reduce some of these knobs...we don't need to show every knob at every level. Right now, the 'activebackgroundshape' knob on the drawer does nothing, because there's a knob for it on the navgroups that overrides it. I would suggest showing the shared prop knobs on the drawer, and the other levels only showing the props specific to them.

Reply:
Yes I agree... I was trying to think of a better way to show how the props can inherit each other. I tried to have two navGroups, but that is equally confusing.